libraries/AP_Mount/AP_Mount_Viewpro.h

```diff
@@ -0,0 +1,310 @@
+        E2 = 0x2E,              // tracking controls infrequently used (sent)
+        T1_F1_B1_D1 = 0x40,     // actual roll, pitch, yaw angles (received)
+    };
+
+    // A1 servo status enum (used in A1, B1 packets)
+    enum class ServoStatus : uint8_t {
+        manual_speed_mode = 0x01,
+        manual_absolute_angle_mode = 0x0B,
+    };
+
+    // C1 image sensor choice
+    enum class ImageSensor : uint8_t {
+        NO_ACTION = 0x00,   // no image sensor is affected
+        EO1 = 0x01,         // electro-optical, aka rgb
+        IR = 0x02,          // infrared, aka thermal
+        EO1_IR_PIP = 0x03,  // rgb is main, IR is picture-in-picture
+        IR_EO1_PIP = 0x04,  // thermal is main, rgb is picture-in-picture
+        FUSION = 0x05,      // rgb and thermal are fused
+        IR1_13MM = 0x06,    // only valid for 1352 module
+        IR2_52MM = 0x07,    // only valid for 1352 module
+    };
+
+    // C1 camera commands
+    enum class CameraCommand : uint8_t {
+        NO_ACTION = 0,
+        STOP_FOCUS_AND_ZOOM = 0x01,
+        ZOOM_OUT = 0x08,
+        ZOOM_IN = 0x09,
+        FOCUS_PLUS = 0x0A,
+        FOCUS_MINUS = 0x0B,
+        TAKE_PICTURE = 0x13,
+        START_RECORD = 0x14,
+        STOP_RECORD = 0x15,
+        AUTO_FOCUS = 0x19,
+        MANUAL_FOCUS = 0x1A
+    };
+
+    // C2 camera commands
+    enum class CameraCommand2 : uint8_t {
+        SET_EO_ZOOM = 0x53
+    };
+
+    // E1 tracking commands
+    enum class TrackingCommand : uint8_t {
+        STOP = 0x01,
+        START = 0x03,
+        SET_POINT = 0x0A,
+        SET_RECT_TOPLEFT = 0x0B,
+        SET_RECT_BOTTOMRIGHT = 0x0C,
+    };
+
+    // E1 tracking source (e.g. which camera)
+    enum class TrackingSource : uint8_t {
+        EO1 = 0x01,         // electro-optical, aka rgb
+        IR = 0x02,          // infrared, aka thermal
+        EO2 = 0x03,         // electro-optical, aka rgb
+    };
+
+    // E2 tracking commands2
+    enum class TrackingCommand2 : uint8_t {
+        SET_POINT = 0x0A,
+        SET_RECT_TOPLEFT = 0x0B,
+        SET_RECT_BOTTOMRIGHT = 0x0C,
+    };
+
+    // parsing state
+    enum class ParseState : uint8_t {
+        WAITING_FOR_HEADER1,
+        WAITING_FOR_HEADER2,
+        WAITING_FOR_HEADER3,
+        WAITING_FOR_LENGTH,
+        WAITING_FOR_FRAMEID,
+        WAITING_FOR_DATA,
+        WAITING_FOR_CRC
+    };
+
+    // packet formats
+    union HandshakePacket {
+        struct PACKED {
+            FrameId frame_id;           // always 0x00
+            uint8_t unused;             // always 0x00
+        } content;
+        uint8_t bytes[sizeof(content)];
+    };
+
+    // A1 used to send target angles and rates
+    union A1Packet {
+        struct PACKED {
+            FrameId frame_id;           // always 0x1A
+            ServoStatus servo_status;   // see ServoStatus enum above
+            uint8_t yaw_msb;            // target yaw angle or rate msb
```

Comment:
uint16_t and use macros to access, name the variable with _be to remind dev you need to convert

Reply:
or add a uint16_be_t class which does auto set/get on use with conversion

Reply:
Thanks for the review.  I've replaced the separate msb,lsb bytes with be16y.